.github/workflows/main-workflow.yaml

```diff
@@ -5,9 +5,24 @@ on:
     branches: [main]
   pull_request:
 
-permissions:
+permissions: 
   contents: write
-
+  actions: read
+  attestations: read
+  checks: read
+  deployments: read
+  discussions: read
+  issues: read
+  packages: read
+  pages: read
+  pull-requests: read
+  repository-projects: read
+  statuses: read
+  security-events: read
+  # read does not seem to exist for id-token
+  id-token: write
+  models: read
```

Comment:
@Yuri05 the main workflow did not run initially
The reason was the the `R-CMD-Check` and `coverage` actions require a `read-all` 
and `bump` and `pkgdown` require `read` for all except `content` that requires `write`

In the end, I had to define them all one by one because of that.
Additionally, the `id-token` permission does not accept a `read` (either `write` or `none` but the actions required `read` at least)
But `id-token` does not have a `read`